src/gravel/controllers/nodes.py

```diff
@@ -171,8 +172,21 @@ async def join(self, address: str, token: str) -> bool:
         joinmsg = JoinMessageModel(uuid=uuid, token=token)
         msg = MessageModel(type=MessageTypeEnum.JOIN, data=joinmsg.dict())
         await conn.send(msg)
+
         reply: MessageModel = await conn.receive()
         logger.debug(f"=> mgr -- join > recv: {reply}")
+        assert reply.type == MessageTypeEnum.WELCOME
+        welcome = WelcomeMessageModel.parse_obj(reply.data)
+        assert welcome.cluster_uuid
+        assert welcome.pubkey
+
+        authorized_keys: Path = Path("/root/.ssh/authorized_keys")
```

Comment:
I don't like this hardcoded authorized keys, or the fact that aquarium will modify the system keys.

However I think the alternatives in the way they would need to interoperate with cephadm are too complex to worry about for now.

Reply:
This is where cephadm is dropping the key during bootstrap, and I could not think of any other (reasonable) way to achieve this. We need that public key to be in the node's authorized_keys so the orchestrator can reach out to the node.

As for changing system keys, I mean... We are an opinionated appliance. We ought to be able to do opinionated things :)

Reply:
> This is where cephadm is dropping the key during bootstrap, and I could not think of any other (reasonable) way to achieve this. We need that public key to be in the node's authorized_keys so the orchestrator can reach out to the node.

Right, that's why I mentioned it'd be difficult to do differently.

> As for changing system keys, I mean... We are an opinionated appliance. We ought to be able to do opinionated things :)

But yes, very good point. :+1: